src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -167,7 +167,7 @@ private BoundBadExpression BadExpression(SyntaxNode syntax, LookupResultKind res
             return new BoundBadExpression(syntax,
                 resultKind,
                 symbols,
-                childNodes,
+                childNodes.SelectAsArray(e => BindToTypeForErrorRecovery(e)),
```

Comment:
>BindToTypeForErrorRecovery [](start = 46, length = 26)

This lambda cannot be cached, which is something we typically try to avoid. Consider either using a foreach loop, or using the `SelectAsArray` overload that allows you pass an extra parameter and pass `this` to allow the lambda to be cached.